### DISCUSSION TITLE
vue-generator这边怎么调试呢？ console.log不会在浏览器打印

### DISCUSSION THREAD

**Discussion by youyanping** [opentiny/tiny-engine#373] [Q&A]
vue-generator这边怎么调试呢？ console.log不会在浏览器打印

**Comment by ghost** [comment]
@youyanping generator是以npm包的形式在mockServer工程中生效，无法直接调试。
vue-generator作为一个独立的工程，有测试用例可以调试。调试步骤如下：
1. 查看test文件夹下测试用例配置：packages\vue-generator\test\testcases\full\index.config.js，保留一条case
![image](https://github.com/opentiny/tiny-engine/assets/142570291/19e22fa4-4fd1-452d-b71f-c107fdea20c1)

2. 将当前页面的schema贴到packages\vue-generator\test\testcases\full\case1_normal\input\page.schema.json
3. 查看依赖的组件列表packages\vue-generator\test\testcases\full\case1_normal\input\components-map.json，如果只依赖@opentiny/vue的组件可不用修改，有第三方组件库组件，需要按格式添加
![image](https://github.com/opentiny/tiny-engine/assets/142570291/c88fd4a3-d55c-45cb-831f-abe1a128c74a)

4. 从入口开始packages\vue-generator\src\generator\page.js，generateCode方法打上断点
![image](https://github.com/opentiny/tiny-engine/assets/142570291/59606f6c-e831-4a87-a187-339ea022a8b7)
5. 打开debug终端
![image](https://github.com/opentiny/tiny-engine/assets/142570291/b504c9f4-5782-410a-b839-056fda595759)

6. 运行npm run test:latest

**Comment by wenmine** [comment]
方案2：
若您没有启动我们的后端工程（webservice），由于我们本地使用的是mockServer作为我们的mock服务器，我们可以开启mockServer的调试模式：
在`mockServer/gulpfile.js`里修改：
![image](https://github.com/opentiny/tiny-engine/assets/14882982/f6505b42-5a30-449c-94c7-c6c61eb66781)

然后在tiny-engine根目录里启动工程`pnpm dev`

然后就能进入后端调试模式，我们可以看到mockServer的文件，找到schema2code.js文件，里面有`generateCode`函数，可以从这里断点打进入到vue-generator包里真正出码的地方。
